src/packagedcode/models.py

```diff
@@ -94,14 +103,80 @@
   -- code in source or compiled form or both.
 """
 
-class Package(object):
+
+class EnhancedModel(Model):
+    def __init__(self, deserialize_mapping=None, strict=True, **kwargs):
+        Model.__init__(self, raw_data=kwargs, deserialize_mapping=deserialize_mapping, strict=strict)
+
+
+class Version(EnhancedModel):
+    version = StringType()
+
+
+class AssertedLicense(EnhancedModel):
+    """
+    As asserted in a package
+    """
+    license = StringType(default=None)
+    text = StringType(default=None)
+    notice = StringType(default=None)
+    url = StringType(default=None)
+
+
+class Party(EnhancedModel):
+    """
+    A party is a person, project or organization
+    """
+    name = StringType()
+    url = StringType()
+    email = StringType()
+    party_person = 'person'
+    # often loosely defined
+    party_project = 'project'
+    # more formally defined
+    party_org = 'organization'
+
+    PARTY_TYPES = (party_person, party_project, party_org,)
+
+    type = StringType(choices=PARTY_TYPES)
+
+
+class Dependency(EnhancedModel):
+    """
+    A dependency points to a Package via a package id and version, and jhas a version
+    constraint  (such as ">= 3.4"). The version is the effective version that
+    has been picked and resolved.
+    """
+    id = StringType()
+    # the effective or concrete resolved and used version
+    version = StringType()
+
+    # the potential constraints for this dep.
+    # this is package format specific
+    version_constraint = StringType()
+
+    # a normalized list of version constraints for this dep.
+    # this is package indepdenent
```

Comment:
There is typo in the comment. But I would need to get some explanation on how you see versioning working first